when more than one SDL2 audio user is active, the first program to call
SDL_CloseAudio() crashes with a segfault inside the hotplug thread
which scans every 100 ms whether a new alsa device is plugged in.
we simply disable the wasteful scanning for new devices while a process runs.
the backtraces obtained differ, one is:

#0  0x00007ffff7d8d4ed in unbin (c=c@entry=0x7ffff7847e70, i=i@entry=1)
    at src/malloc/malloc.c:193
#1  0x00007ffff7d8de72 in malloc (n=<optimized out>, n@entry=6)
    at src/malloc/malloc.c:320
#2  0x00007fffefedec3a in copy_local_string (s=s@entry=0x7ffff7f5c508)
    at conf.c:955
#3  0x00007fffefedfb5f in get_freestring (input=0x7ffff7f5c750, id=0, 
    string=0x7ffff7f5c5a8) at conf.c:1002
#4  get_string (string=string@entry=0x7ffff7f5c5a8, id=id@entry=0, 
    input=input@entry=0x7ffff7f5c750) at conf.c:1081
#5  0x00007fffefedfbe2 in parse_value (_n=_n@entry=0x7ffff7f5c618, 
    parent=parent@entry=0x1354020, input=input@entry=0x7ffff7f5c750, 
    id=id@entry=0x7ffff7f5c610, skip=skip@entry=0) at conf.c:1152
#6  0x00007fffefee14b9 in parse_def (override=0, skip=<optimized out>, 
    input=0x7ffff7f5c750, parent=0x1354020) at conf.c:1438
#7  parse_defs (parent=0x1354020, input=input@entry=0x7ffff7f5c750, 
    skip=skip@entry=0, override=override@entry=0) at conf.c:1466
#8  0x00007fffefee15ff in parse_array_def (override=0, skip=0, idx=0, 
    input=0x7ffff7f5c750, parent=0xbe9c40) at conf.c:1263
#9  parse_array_defs (parent=0xbe9c40, input=input@entry=0x7ffff7f5c750, 
    skip=0, override=override@entry=0) at conf.c:1305
#10 0x00007fffefee145c in parse_def (override=0, skip=<optimized out>, 
    input=0x7ffff7f5c750, parent=0xa46840) at conf.c:1424
#11 parse_defs (parent=parent@entry=0xa46840, 
    input=input@entry=0x7ffff7f5c750, skip=skip@entry=0, 
    override=override@entry=0) at conf.c:1466
#12 0x00007fffefee17ea in _snd_config_load_with_include (config=0xa46840, 
    in=<optimized out>, override=override@entry=0, 
    default_include_path=default_include_path@entry=0x0) at conf.c:1862
#13 0x00007fffefee1956 in snd_config_load (config=<optimized out>, 
    in=<optimized out>) at conf.c:1931
#14 0x00007fffefee4331 in snd_config_update_r (_top=_top@entry=0x7ffff7f5c8f0, 
    _update=_update@entry=0x7ffff7f5c900, cfgs=cfgs@entry=0x0) at conf.c:4013
#15 0x00007fffefee9794 in snd_device_name_hint (card=<optimized out>, 
    iface=0x7ffff7b54604 "pcm", hints=0x7ffff7f5ca10) at namehint.c:560
#16 0x00007ffff7b0dd0d in ALSA_HotplugThread (arg=arg@entry=0xa09e80)
    at /src/build/sdl2/SDL2-2.0.16/src/audio/alsa/SDL_alsa_audio.c:802
#17 0x00007ffff7ac7125 in SDL_RunThread (thread=0xbe9f20)
    at /src/build/sdl2/SDL2-2.0.16/src/thread/SDL_thread.c:275
#18 0x00007ffff7b29835 in RunThread (data=<optimized out>)
    at /src/build/sdl2/SDL2-2.0.16/src/thread/pthread/SDL_systhread.c:77
#19 0x00007ffff7dbcd4e in start (p=0x7ffff7f5cab0)
    at src/thread/pthread_create.c:192
#20 0x00007ffff7dbeede in __clone () at src/thread/x86_64/clone.s:22
Backtrace stopped: frame did not save the PC


--- SDL2-2.0.16.org/src/audio/alsa/SDL_alsa_audio.c	2021-09-06 19:52:19.492449213 +0000
+++ SDL2-2.0.16/src/audio/alsa/SDL_alsa_audio.c	2021-09-13 19:19:11.659527811 +0000
@@ -920,8 +920,8 @@
         }
 
         /* Block awhile before checking again, unless we're told to stop. */
-        ticks = SDL_GetTicks() + 5000;
-        while (!SDL_AtomicGet(&ALSA_hotplug_shutdown) && !SDL_TICKS_PASSED(SDL_GetTicks(), ticks)) {
+        //ticks = SDL_GetTicks() + 5000;
+        while (!SDL_AtomicGet(&ALSA_hotplug_shutdown)) {
             SDL_Delay(100);
         }
     }
